DllImportGenerator/DllImportGenerator.UnitTests/Compiles.cs

```diff
@@ -33,21 +33,39 @@ public static IEnumerable<object[]> CodeSnippetsToCompile_NoDiagnostics()
             yield return new[] { CodeSnippets.BasicParametersAndModifiers<bool>() };
             yield return new[] { CodeSnippets.BasicParametersAndModifiers<IntPtr>() };
             yield return new[] { CodeSnippets.BasicParametersAndModifiers<UIntPtr>() };
-            //yield return new[] { CodeSnippets.BasicParametersAndModifiers<byte[]>() };
-            //yield return new[] { CodeSnippets.BasicParametersAndModifiers<sbyte[]>() };
-            //yield return new[] { CodeSnippets.BasicParametersAndModifiers<short[]>() };
-            //yield return new[] { CodeSnippets.BasicParametersAndModifiers<ushort[]>() };
-            //yield return new[] { CodeSnippets.BasicParametersAndModifiers<int[]>() };
-            //yield return new[] { CodeSnippets.BasicParametersAndModifiers<uint[]>() };
-            //yield return new[] { CodeSnippets.BasicParametersAndModifiers<long[]>() };
-            //yield return new[] { CodeSnippets.BasicParametersAndModifiers<ulong[]>() };
-            //yield return new[] { CodeSnippets.BasicParametersAndModifiers<float[]>() };
-            //yield return new[] { CodeSnippets.BasicParametersAndModifiers<double[]>() };
-            //yield return new[] { CodeSnippets.BasicParametersAndModifiers<bool[]>() };
-            //yield return new[] { CodeSnippets.BasicParametersAndModifiers<char[]>() };
-            //yield return new[] { CodeSnippets.BasicParametersAndModifiers<string[]>() };
-            //yield return new[] { CodeSnippets.BasicParametersAndModifiers<IntPtr[]>() };
-            //yield return new[] { CodeSnippets.BasicParametersAndModifiers<UIntPtr[]>() };
+            yield return new[] { CodeSnippets.ArrayParametersAndModifiers<byte>() };
+            yield return new[] { CodeSnippets.ArrayParametersAndModifiers<sbyte>() };
+            yield return new[] { CodeSnippets.ArrayParametersAndModifiers<short>() };
+            yield return new[] { CodeSnippets.ArrayParametersAndModifiers<ushort>() };
+            yield return new[] { CodeSnippets.ArrayParametersAndModifiers<int>() };
+            yield return new[] { CodeSnippets.ArrayParametersAndModifiers<uint>() };
+            yield return new[] { CodeSnippets.ArrayParametersAndModifiers<long>() };
+            yield return new[] { CodeSnippets.ArrayParametersAndModifiers<ulong>() };
+            yield return new[] { CodeSnippets.ArrayParametersAndModifiers<float>() };
+            yield return new[] { CodeSnippets.ArrayParametersAndModifiers<double>() };
+            yield return new[] { CodeSnippets.ArrayParametersAndModifiers<bool>() };
+            yield return new[] { CodeSnippets.ArrayParametersAndModifiers<IntPtr>() };
+            yield return new[] { CodeSnippets.ArrayParametersAndModifiers<UIntPtr>() };
+            yield return new[] { CodeSnippets.ArrayParameterWithSizeParam<byte>(false) };
```

Comment:
Let's use the argument name with free boolean constants here. Or perhaps we can leverage an existing enum to indicate the byref state?